rllib/agents/callbacks.py

```diff
@@ -381,3 +385,59 @@ def on_learn_on_batch(self, *, policy: Policy, train_batch: SampleBatch,
     def on_train_result(self, *, trainer, result: dict, **kwargs) -> None:
         for callback in self._callback_list:
             callback.on_train_result(trainer=trainer, result=result, **kwargs)
+
+
+# This Callback is used by the RE3 exploration strategy.
```

Comment:
Could we move this into `utils/exploration/random_encoder.py`?

Reply:
It was moved out from here to avoid circular dependency in this commit https://github.com/ray-project/ray/pull/19551/commits/89169ccb63b0a780e3eb6231a38907244cfc1278

Reply:
yeah, this is the best I could do.
I would love to keep everything in random_encoder.py too, but you know :)